pkg/api/handlers/libpod/manifests.go

```diff
@@ -207,3 +275,155 @@ func ManifestPush(w http.ResponseWriter, r *http.Request) {
+		return
+	}
+	defer auth.RemoveAuthfile(authfile)
+	var username, password string
+	if authconf != nil {
+		username = authconf.Username
+		password = authconf.Password
+	}
+	options := entities.ImagePushOptions{
+		Authfile: authfile,
+		Username: username,
+		Password: password,
+		All:      query.All,
+	}
+	if sys := runtime.SystemContext(); sys != nil {
+		options.CertDir = sys.DockerCertPath
+	}
+	if _, found := r.URL.Query()["tlsVerify"]; found {
+		options.SkipTLSVerify = types.NewOptionalBool(!query.TLSVerify)
+	}
+
+	imageEngine := abi.ImageEngine{Libpod: runtime}
+	source := utils.GetName(r)
+	digest, err := imageEngine.ManifestPush(context.Background(), source, destination, options)
+	if err != nil {
+		utils.Error(w, "Something went wrong.", http.StatusBadRequest, errors.Wrapf(err, "error pushing image %q", destination))
+		return
+	}
+	utils.WriteResponse(w, http.StatusOK, handlers.IDResponse{ID: digest})
+}
+
+// ManifestModify efficiently updates the named manifest list
+func ManifestModify(w http.ResponseWriter, r *http.Request) {
+	runtime := r.Context().Value(api.RuntimeKey).(*libpod.Runtime)
+	imageEngine := abi.ImageEngine{Libpod: runtime}
+
+	body := new(entities.ManifestModifyOptions)
+	if err := json.NewDecoder(r.Body).Decode(body); err != nil {
+		utils.Error(w, "Something went wrong.", http.StatusInternalServerError, errors.Wrap(err, "Decode()"))
+		return
+	}
+
+	name := utils.GetName(r)
+	if _, err := runtime.LibimageRuntime().LookupManifestList(name); err != nil {
+		utils.Error(w, "Something went wrong.", http.StatusNotFound, err)
+		return
+	}
+
+	var report entities.ManifestModifyReport
+	switch {
+	case strings.EqualFold("update", body.Operation):
+		id, err := imageEngine.ManifestAdd(r.Context(), name, body.Images, body.ManifestAddOptions)
+		if err != nil {
+			utils.InternalServerError(w, err)
+			return
+		}
+		report = entities.ManifestModifyReport{
+			ID:     id,
+			Images: body.Images,
+		}
+	case strings.EqualFold("remove", body.Operation):
+		report := entities.ManifestModifyReport{
+			Images: make([]string, 0),
+		}
+		for _, image := range body.Images {
+			id, err := imageEngine.ManifestRemoveDigest(r.Context(), name, image)
+			if err != nil {
+				logrus.Infof("Failed to remove digest %q from manifest %q", image, name)
+				continue
+			}
+			report.ID = id
+			report.Images = append(report.Images, image)
+		}
+	case strings.EqualFold("annotate", body.Operation):
+		options := entities.ManifestAnnotateOptions{
+			Annotation: body.Annotation,
+			Arch:       body.Arch,
+			Features:   body.Features,
+			OS:         body.OS,
+			OSFeatures: body.OSFeatures,
+			OSVersion:  body.OSVersion,
+			Variant:    body.Variant,
+		}
+		report := entities.ManifestModifyReport{
+			Images: make([]string, 0),
+		}
+		for _, image := range body.Images {
+			id, err := imageEngine.ManifestAnnotate(r.Context(), name, image, options)
+			if err != nil {
```

Comment:
Shouldn't Errors be returned in report.Err?

Reply:
Good catch